### PR TITLE
[ci][configs] make ray project docker name configurable

### PR DESCRIPTION
**Pull request by can-anyscale**
Move ray project docker name to be configurable.

Test strategies:
- Release test

### CODE REVIEW

File: release/ray_release/configs/global_config.py [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 class GlobalConfig(TypedDict):
     byod_ray_ecr: str
+    byod_ray_cpu_project: str
+    byod_ray_gpu_project: str
```
Comment (aslonnie) [inline release/ray_release/configs/global_config.py:8]: this should be named as `ray_ml` not `ray_gpu`.

`rayproject/ray` also has `gpu` tag. `gpu` tag means that the container is based on some version of `cuda` image. `ml` tag means that a big list of additional apt and python packages are installed in the image.

File: release/ray_release/configs/global_config.py [hunk 2]
```diff
@@ -4,6 +4,8 @@
 
 class GlobalConfig(TypedDict):
     byod_ray_ecr: str
+    byod_ray_project: str
```
Comment (aslonnie) [inline release/ray_release/configs/global_config.py:7]: could you change it to maybe `ray_cr_repo` and `ray_ml_cr_repo`?

"project" is used as a suffix on github https://github.com/ray-project and docker hub https://hub.docker.com/u/rayproject , which is part of the organization name.

File: release/ray_release/test.py [hunk 3]
```diff
@@ -255,9 +255,14 @@ def get_ray_image(self) -> str:
         """
         Returns the ray docker image to use for this test.
         """
-        ray_project = "ray" if self.get_byod_type() == "cpu" else "ray-ml"
+        config = get_global_config()
+        ray_project = (
+            config["byod_ray_project"]
+            if self.get_byod_type() == "cpu"
```
[the remark this answers is not in the document]
Comment (can-anyscale) [reply release/ray_release/test.py:261]: do you mean the byod type? 100% agree, I can rework the byod type in a later PR to keep this one small

File: release/ray_release/configs/global_config.py (hunk 2 above)
Comment (can-anyscale) [reply release/ray_release/configs/global_config.py:7]: totally

**Review by aslonnie — APPROVED** [review]

**Review by krfricke — APPROVED** [review]

**Pull request merged**